src/components/GoalDependencyForm/GoalDependencyForm.tsx

```diff
@@ -0,0 +1,113 @@
+import { GoalSuggest } from '../GoalSuggest';
+import { InlineForm } from '../InlineForm';
+import { InlineTrigger } from '../InlineTrigger';
+
+import { tr } from './GoalDependencyForm.i18n';
+
+const StyledFormInput = styled(FormInput)`
+    font-size: 14px;
+    font-weight: normal;
+    padding: 5px 10px;
+
+    border: 1px solid ${gray7};
+    border-right: 0;
+    box-sizing: border-box;
+`;
+
+const StyledInlineTrigger = styled(InlineTrigger)`
+    line-height: 28px;
+`;
+
+interface GoalDependencyAddFormProps {
+    goalId: string;
+    kind: dependencyKind;
+    onSubmit: (values: ToggleGoalDependency) => void;
+}
+
+export const GoalDependencyAddForm: React.FC<GoalDependencyAddFormProps> = ({ goalId, kind, onSubmit }) => {
+    const [selected, setSelected] = useState<GoalByIdReturnType | null>(null);
+    const [query, setQuery] = useState(['']);
+    const {
+        control,
+        handleSubmit,
+        setValue,
+        reset,
+        formState: { isSubmitSuccessful },
+    } = useForm({
+        resolver: zodResolver(toggleGoalDependencySchema),
+        mode: 'onChange',
+        defaultValues: {
+            relation: { id: '' },
+            id: goalId,
+            kind,
+        },
+    });
+
+    const resetFormHandler = useCallback(() => {
+        setSelected(null);
+        setQuery(['']);
+
+        reset({
+            relation: { id: '' },
+            id: goalId,
+            kind,
+        });
+    }, [reset, goalId, kind]);
+
+    useEffect(() => {
+        if (selected) {
+            setValue('relation.id', selected.id);
+        }
+    }, [selected, setValue]);
+
+    const handleInputChange = useCallback<React.ChangeEventHandler<HTMLInputElement>>((event) => {
+        setQuery([event.target.value]);
+    }, []);
+
+    const handleGoalSelect = useCallback((goal: NonNullable<GoalByIdReturnType>) => {
+        setSelected(goal);
+        setQuery([goal.projectId!, String(goal.scopeId)]);
+    }, []);
+
+    return (
+        <InlineForm
+            renderTrigger={(props) => (
+                <StyledInlineTrigger text={tr('Add dependency')} icon={<PlusIcon noWrap size="s" />} {...props} />
+            )}
+            submitted={isSubmitSuccessful}
+            onSubmit={handleSubmit(onSubmit)}
+            reset={resetFormHandler}
+        >
+            <GoalSuggest
+                renderTrigger={() => (
+                    <StyledFormInput
+                        autoFocus
+                        autoComplete="off"
+                        value={query.join('-')}
+                        onChange={handleInputChange}
+                        brick="right"
+                    />
+                )}
+                value={query.join('-')}
+                onChange={handleGoalSelect}
+            />
+            <Button text={tr('Add')} brick="left" type="submit" view="primary" outline disabled={selected == null} />
+            <Controller
+                name="relation"
```

Comment:
Use `register` https://github.com/taskany-inc/issues/blob/main/src/components/CommentForm/CommentForm.tsx#L120

Reply:
for what? I can pass the name as `relation.id` and `control` will return a correct value into render props function